apps/docs/pages/docs/api-reference/fields/richtext.mdx

```diff
@@ -0,0 +1,733 @@
+| [`selector`](#selector)                   | `selector: (ctx) => ({ ... })`                                                      | `RichTextSelector`                                             | -        |
+
+## Required params
+
+### `type`
+
+The type of the field. Must be `"richtext"` for RichText fields.
+
+```tsx {6} copy
+const config = {
+  components: {
+    Example: {
+      fields: {
+        description: {
+          type: "richtext",
+        },
+      },
+      // ...
+    },
+  },
+};
+```
+
+## Optional params
+
+### `contentEditable`
+
+Enable inline text editing for this field.
+
+```tsx {7} copy
+const config = {
+  components: {
+    Example: {
+      fields: {
+        description: {
+          type: "richtext",
+          contentEditable: false,
+        },
+      },
+      // ...
+    },
+  },
+};
+```
+
+### `menu`
+
+Provide an override for the grouped button layout rendered in the editor toolbar. Each key in your `menu` matches a menu group and maps to an array of control keys.
+
+```tsx {7-10} copy
+const config = {
+  components: {
+    Example: {
+      fields: {
+        description: {
+          type: "richtext",
+          menu: {
+            typography: ["Bold", "Italic", "Underline"],
+            blocks: ["BulletList", "OrderedList"],
+          },
+        },
+      },
+    },
+  },
+};
+```
+
+### `inlineMenu`
+
+Provide an override for the grouped button layout rendered in the `BubbbleMenu` when inline editing of your field. Each key in your `inlineMenu` matches a menu group and maps to an array of control keys.
+
+```tsx {7-10} copy
+const config = {
+  components: {
+    Example: {
+      fields: {
+        description: {
+          type: "richtext",
+          inlineMenu: {
+            typography: ["Bold", "Italic", "Underline"],
+            blocks: ["BulletList", "OrderedList"],
+          },
+        },
+      },
+    },
+  },
+};
+```
+
+### `controls`
+
+Controls let you swap or extend the buttons available to each menu key. A control is an object with a `render(editor, editorState)` function, so you can plug in any React UI while still receiving the merged TipTap state produced by our selector utilities.
+
+```tsx {1-12, 20-22} copy
+const BoldControl = {
+  render: (editor, editorState) => (
+    <button
+      type="button"
+      onClick={(e) => {
+        e.stopPropagation();
+        editor.chain().focus().toggleBold().run();
+      }}
+      disabled={!editorState.canBold}
+      style={{ fontWeight: editorState.isBold ? "bold" : "normal" }}
+    >
+      Bold
+    </button>
+  ),
+};
+
+const config = {
+  components: {
+    Example: {
+      fields: {
+        description: {
+          type: "richtext",
+          controls: {
+            Bold: BoldControl,
+          },
+        },
+      },
+    },
+  },
+};
+```
+
+<Callout>
+  Note: The `e.stopPropagation();` is required in Puck in order to avoid issues
+  with the portal on inlineMenu.
+</Callout>
+
+### `textSelectOptions`
```

Comment:
The user is able to set tags regardless what value is provided to textSelectOptions.

For example, using markdown, the user can set an H1. But because "textSelectOptions" doesn't contain an H1, it shows "Paragraph".

I haven't tested it, but I imagine that this might extend beyond this API. I.e. if the user sets bold but there is no bold control.

We likely need to tweak the API here. As a developer, I would expect to be able to control exactly what tags are available in my RTE. I would also expect Puck to automatically show the controls (and adjust values in the text select) based on what tags I allow.

So something like this: `validTags: ["h1", "h2", "p", "b"]`.

Reply:
Don't totally follow.  Lets discuss